pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -542,14 +542,25 @@ func ResolvePipelineTask(
 	getTaskRun resources.GetTaskRun,
 	getRun GetRun,
 	pipelineTask v1.PipelineTask,
+	pst PipelineRunState,
 ) (*ResolvedPipelineTask, error) {
 	rpt := ResolvedPipelineTask{
 		PipelineTask: &pipelineTask,
 	}
 	rpt.CustomTask = rpt.PipelineTask.TaskRef.IsCustomTask() || rpt.PipelineTask.TaskSpec.IsCustomTask()
 	numCombinations := 1
+	// We want to resolve all of the result references and ignore any errors at this point since there could be
+	// instances where result references are missing here, but will be later skipped or resolved in a subsequent
+	// TaskRun. The final validation is handled in skipBecauseResultReferencesAreMissing.
+	resolvedResultRefs, _, _ := ResolveResultRefs(pst, PipelineRunState{&rpt})
+	if err := validateArrayResultsIndex(resolvedResultRefs); err != nil {
+		return nil, err
```

Comment:
nit (not a block for the PR): I think this line is not covered?

Reply:
For this PR the coverage for `ValidateArrayResultsIndex()` exists within `resolutionrefresolution_test.go`.

https://github.com/tektoncd/pipeline/blob/b5d6047bf924006345f9599be42640fd1b758595/pkg/reconciler/pipelinerun/resources/resultrefresolution_test.go#L619

However I do have a pipelinerun.go unit test that covers an invalid array result index in  the whole array results implementation https://github.com/tektoncd/pipeline/blob/c164f71ab448c4ac263d0113bf1b58d1c0199be3/pkg/reconciler/pipelinerun/pipelinerun_test.go#LL10603C6-L10603C66

Reply:
The test in TestValidateArrayResultsIndex only tests the `validateArrayResultsIndex` but not tests the `ResolvePipelineTask` where we call the function. 

The 2nd link makes sense! The validation happened in `pipelinerun.go` before but in this PR we moved it to the `ResolvePipelineTask`, I think it should be ok to rely on that test.

This line is red in the coverage report but I think that's fine.